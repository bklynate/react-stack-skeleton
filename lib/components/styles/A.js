'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = A;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The nav bar for the website.
 */
function A(_ref) {
  var to = _ref.to,
      href = _ref.href,
      label = _ref.label,
      otherProps = _objectWithoutProperties(_ref, ['to', 'href', 'label']);

  if (to) return _react2.default.createElement(
    _reactRouter.Link,
    { to: to },
    label
  );else if (href) return _react2.default.createElement(
    'a',
    { href: href },
    label
  );else return _react2.default.createElement(
    'a',
    { href: '#' },
    label
  );
}