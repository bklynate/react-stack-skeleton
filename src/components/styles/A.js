/* @flow */
import React from 'react';
import { Link } from 'react-router';

import type { NavItem } from '../../stores/navigation';

type Props = NavItem;

/**
 * The nav bar for the website.
 */
export default function A({ to, href, label, ...otherProps }: Props): React.Element<*> {
  if(to)        return <Link to={to}>{label}</Link>;
  else if(href) return <a href={href}>{label}</a>;
  else          return <a href="#">{label}</a>;
}
