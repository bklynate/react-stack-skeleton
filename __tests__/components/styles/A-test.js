import React from 'react';
import renderer from 'react-test-renderer';

import A from '../../../src/components/styles/A';


test('The A component displays the given label.', () => {
  const component = renderer.create(
    <A label="Test Link" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('The A component displays a normal anchore tag that goes to "#" when neither "href" nor "to" props is given.', () => {
  const component = renderer.create(
    <A />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('The A component displays a normal anchore tag when given a "href" prop.', () => {
  const component = renderer.create(
    <A href="https://google.com" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('The A component displays a react-router Link as the anchore tag when given a "to" prop.', () => {
  const component = renderer.create(
    <A to="/" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
