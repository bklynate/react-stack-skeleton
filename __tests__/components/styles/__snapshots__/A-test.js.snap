exports[`test The A component displays a normal anchore tag that goes to "#" when neither "href" nor "to" props is given. 1`] = `
<a
  href="#" />
`;

exports[`test The A component displays a normal anchore tag when given a "href" prop. 1`] = `
<a
  href="https://google.com" />
`;

exports[`test The A component displays a react-router Link as the anchore tag when given a "to" prop. 1`] = `
<a
  onClick={[Function]}
  style={Object {}} />
`;

exports[`test The A component displays the given label. 1`] = `
<a
  href="#">
  Test Link
</a>
`;
